pyth_observer/crosschain.py

```diff
@@ -0,0 +1,27 @@
+import requests
+
+from throttler import throttle
+
+
+# We have an attester that sends the prices to wormhole to be verified.
+# Currently it sends prices to wormhole once every minute due to costs.
+# We can increase the frequency of the attester once we're live on pythnet.
+@throttle(rate_limit=1, period=60)
+async def get_crosschain_prices(cluster):
+    price_feed_ids = requests.get(
+        f"https://xc-{cluster}.pyth.network/api/price_feed_ids"
+    ).json()
+    query_params = "ids[]=" + "&ids[]=".join(price_feed_ids)
+    latest_price_feeds = requests.get(
+        f"https://xc-{cluster}.pyth.network/api/latest_price_feeds?{query_params}"
+    ).json()
```

Comment:
Couple comments on this url configuration strategy:
* this is a relatively inflexible way to configure the URLs. What happens if in the future we want to use a URL that doesn't fit this string format? There's no way to handle that case without editing the code.
* you've distributed the configuration across two locations. The `if args.network == "devnet"` check in the file above is conceptually part of the configuration for how to get crosschain prices, but it's in a different file.

A better way to handle this configuration would be to make a class like `CrosschainPriceObserver` that takes the relevant config urls as arguments to the constructor, then exposes `get_crosschain_prices()` as a function on it. You can then configure the urls arbitrarily, and you could also make a version of the same interface that was a noop to handle the case where the network isn't currently supported.

Reply:
makes sense, thanks for the detailed feedback and suggestion -- have pushed a commit that:
- made a `CrosschainPriceObserver` class that takes in a URL endpoint as arg to the constructor
- checks if the endpoint is valid
- exposes `self.valid` attr so that we can call `get_crosschain_prices()` only if endpoint is valid

let me know what are your thoughts on this approach